### PR TITLE
Add headless oauth login endpoint for CLI

### PR DESCRIPTION
**Pull request by alphadose**

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -154,6 +168,80 @@ func IsOauthUser(user *models.User) error {
...
 
+// HandleHeadlessSSO - handles the OAuth login flow for headless interfaces such as Netmaker CLI via websocket
+func HandleHeadlessSSO(w http.ResponseWriter, r *http.Request) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		logger.Log(0, "error during connection upgrade for node sign-in:", err.Error())
+		return
+	}
+	if conn == nil {
+		logger.Log(0, "failed to establish web-socket connection during node sign-in")
```
Comment (0xdcarns) [inline auth/auth.go:179]: errors should probably indicate a CLI or headless sign-in attempt, not node

File: auth/headless_callback.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		w.Write([]byte("bad conf"))
+		logger.Log(0, "Missing Oauth config in HandleHeadlessSSOCallback")
+		return
+	}
+	state, code := getStateAndCode(r)
+
+	userClaims, err := functions[get_user_info].(func(string, string) (*OAuthUser, error))(state, code)
+	if err != nil {
+		logger.Log(0, "error when getting user info from callback:", err.Error())
+		http.Redirect(w, r, servercfg.GetFrontendURL()+"/login?oauth=callback-error", http.StatusTemporaryRedirect)
```
Comment (0xdcarns) [inline auth/headless_callback.go:29]: if this is headless, we need to be wary of a frontend URL not being present

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: auth/auth.go (hunk 1 above)
Comment (alphadose) [reply auth/auth.go]: done

File: auth/headless_callback.go (hunk 2 above)
Comment (alphadose) [reply auth/headless_callback.go]: done

**Review by 0xdcarns — APPROVED** [review]

**Review by abhishek9686 — APPROVED** [review]

**Review by Aceix — APPROVED** [review]

**Pull request merged**